pkg/storage/tsdb/config.go

```diff
@@ -241,14 +245,15 @@ func (cfg *TSDBConfig) RegisterFlags(f *flag.FlagSet) {
 }
 
 // Validate the config.
-func (cfg *TSDBConfig) Validate() error {
+func (cfg *TSDBConfig) Validate(logger log.Logger) error {
 	if cfg.ShipInterval > 0 && cfg.ShipConcurrency <= 0 {
 		return errInvalidShipConcurrency
 	}
 
-	if cfg.MaxTSDBOpeningConcurrencyOnStartup <= 0 {
+	if cfg.DeprecatedMaxTSDBOpeningConcurrencyOnStartup <= 0 {
 		return errInvalidOpeningConcurrency
 	}
+	util.WarnDeprecatedConfig(maxTSDBOpeningConcurrencyOnStartupFlag, logger)
```

Comment:
@pracucci I was thinking about this, but what concerns me is the following: even if an user doesn't set this, the default value will be there and will be used, and therefore the user will indirectly use this deprecated configuration. 
There are 2 possible scenarios:
- I am a user and I don't know that `-blocks-storage.tsdb.max-tsdb-opening-concurrency-on-startup` exists, so I am using it, and whether it is deprecated or not doesn't make any difference for me because I don't know it is there, and 
- I am a user, I know that `-blocks-storage.tsdb.max-tsdb-opening-concurrency-on-startup` exists, and I am OK with its default value. So I am using something that is deprecated, but I don't know it is deprecated.
Once we remove `-blocks-storage.tsdb.max-tsdb-opening-concurrency-on-startup`, the second user will not even know that he is now using, let's say 4 instead of 10 concurrent processes. 
Anyway, I changed my implementation and will log this warning only if the value is different from the default one.

Reply:
That's correct, but also true for any other config we deprecate. We're used to warn about it only for deprecated config for which you change the default value, and I would do it here too.